webapp/src/Controller/API/JudgehostController.php

```diff
@@ -1044,10 +1052,18 @@ private function addSingleJudgingRun(
                 ->setOutputSystem(base64_decode($outputSystem))
                 ->setMetadata(base64_decode($metadata));
 
+            if ($judging->getRejudgingid() !== null && $judging->getRejudging()->getAutoApply()) {
+                $judging->getSubmission()->setRejudging(null);
+                foreach ($judging->getSubmission()->getJudgings() as $j) {
+                    $j->setValid(false);
+                }
+                $judging->setValid(true);
+            }
+
             $this->em->persist($judgingRun);
             $this->em->flush();
 
-            if ($judging->getRejudgingid() === null) {
+            if ($judging->getValid()) {
```

Comment:
Just wondering why we don't have the  `!$this->config->get('verification_required') &&` clause here as above.

Reply:
Because one is judging the other is judging_run.